### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[sc-8981]

File: src/eip1559.ts [hunk 1]
```diff
@@ -74,16 +65,16 @@ const estimatePriorityFee = async (
       return acc;
     }
     const next = arr[i + 1];
-    const p = next.minus(cur).dividedBy(cur).multipliedBy(100);
+    const p = ((next - cur) / cur) * 100n;
     return [...acc, p];
-  }, [] as BigNumber[]);
-  const highestIncrease = BigNumber.max(...percentageIncreases);
-  const highestIncreaseIndex = percentageIncreases.findIndex((p) => p.eq(highestIncrease));
+  }, [] as bigint[]);
```
Comment (Mrtenz) [inline src/eip1559.ts]: Use `reduce` generic parameter instead.

**Review by Mrtenz — APPROVED** [review]

**Review by Mrtenz — CHANGES_REQUESTED** [review]
Did not mean to approve

File: src/eip1559.ts (hunk 1 above)
Comment (FrederikBolding) [reply src/eip1559.ts]: Fixed!

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**